controllers/clusters/kafka_controller.go

```diff
@@ -265,9 +267,180 @@ func (r *KafkaReconciler) handleUpdateCluster(
 		return models.ReconcileRequeue
 	}
 
+	for _, ref := range k.Spec.UserRefs {
+		err = r.handleCreateUser(ctx, k, l, ref)
+		if err != nil {
+			l.Error(err, "Cannot create Kafka user", "user", ref)
+			r.EventRecorder.Eventf(k, models.Warning, models.CreatingEvent,
+				"Cannot create user. Reason: %v", err)
+		}
+	}
+
 	return models.ExitReconcile
 }
 
+func (r *KafkaReconciler) handleCreateUser(
+	ctx context.Context,
+	kafka *v1beta1.Kafka,
+	l logr.Logger,
+	userRef *v1beta1.UserReference,
+) error {
+	req := types.NamespacedName{
+		Namespace: userRef.Namespace,
+		Name:      userRef.Name,
+	}
+
+	u := &kafkamanagementv1beta1.KafkaUser{}
+	err := r.Get(ctx, req, u)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			l.Info("kafka user is not found", "request", req)
+			r.EventRecorder.Event(kafka, models.Warning, "Not Found",
+				"Kafka user is not found, create a new one or provide correct userRef")
+
+			return err
+		}
+
+		l.Error(err, "Cannot get Kafka user", "user", u.Spec)
+		r.EventRecorder.Eventf(kafka, models.Warning, models.CreationFailed,
+			"Cannot get Kafka user. user reference: %v", userRef)
+
+		return err
+	}
+
+	if _, exist := u.Status.ClustersEvents[kafka.Status.ID]; exist {
+		l.Info("User is already existing on the cluster",
```

Comment:
`User is already existing on the cluster` -> `User is already created for the cluster`

Reply:
done